src/cmd/build.rs

```diff
@@ -1173,21 +1200,36 @@ mod tests_ci_only {
     #[test]
     fn keep_debug_symbols_in_release_mode() {
         with_new_contract_project(|manifest_path| {
-            let res = super::execute(
-                &manifest_path,
-                Verbosity::Default,
-                BuildMode::Release,
-                BuildArtifacts::CodeOnly,
-                UnstableFlags::default(),
-                OptimizationPasses::default(),
-                true,
-            )
-            .expect("build failed");
+            let args = crate::cmd::build::ExecuteArgs {
+                manifest_path,
+                build_mode: BuildMode::Release,
+                build_artifact: BuildArtifacts::CodeOnly,
+                keep_debug_symbols: true,
+                ..Default::default()
+            };
+
+            let res = super::execute(args).expect("build failed");
 
             // we specified that debug symbols should be kept
             assert!(has_debug_symbols(&res.dest_wasm.unwrap()));
 
             Ok(())
         })
     }
+
+    #[test]
+    fn build_result_seralization_sanity_check() {
```

Comment:
Tbh I couldn't come up with good tests to figure out whether or not this flag worked as expected. They basically ended up boiling down to whether or not `BuildResult` had the right fields. 

Open to suggestions though.